### DISCUSSION TITLE
Buderus GB125 missing values

### DISCUSSION THREAD

**Discussion by Marcwa19197** [emsesp/EMS-ESP32#986] [General]
**Question**
first of all: Thank you for the great product!
I installed it a few days ago and im able to get some data using MQTT, Telegraf and Grafana with InfluxDB.

But im missing a few values. I have a Buderus GB125 with Logano LT160 as water storage.

Im missing the following values, which should be available in the MQTT Topic "boiler_data":
- reTemp (Im able to see the curFlowTemp but not the Returning Temp)
- sysPress
- pumpStatus
- valveStatus (i have one Heating Cycle, so i subscribed to topic "mixer_data_hc1" and "mixer_data_ww1")

.. or is it because of missing "hardware"?

**Device information**
```
EMS-ESP Version
v3.5.0b9

Device (Platform / SDK)
ESP32 / v4.4.2

System Uptime
000+00:15:10.247

CPU Frequency
240 MHz

Heap (Free / Max Alloc)
123 KB / 39 KB

Flash Chip (Size / Speed)
16,384 KB / 40 MHz

Application (Used / Free)
1,899 KB / 6,229 KB

File System (Used / Free)
20 KB / 44 KB
```

**Additional context**
I subscribed to the following topics via telegraf.conf:
```
"ems-esp/shower_data",
"ems-esp/mixer_data_wwc1",
"ems-esp/mixer_data_hc1",
"ems-esp/thermostat_data",
"ems-esp/boiler_data_info",
"ems-esp/boiler_data_ww",
"ems-esp/boiler_data",
"ems-esp/heating_active",
"ems-esp/tapwater_active",
"ems-esp/heartbeat",

```

Im able to see the following data using telnet to the ems-esp device and running "show" command:
```
Boiler: GB125/MC10 (DeviceID:0x08, ProductID:72, Version:02.07) (61)
  heating active: off
  tapwater active: off
  selected flow temperature: 11.0 °C
  burner selected max power: 100 %
  heating pump modulation: 0 %
  outside temperature: 10.0 °C
  current flow temperature: 70.9 °C
  actual boiler temperature: 71.0 °C
  exhaust temperature: 53.2 °C
  gas: off
  gas stage 2: off
  flame current: 0.0 uA
  heating pump: off
  fan: off
  ignition: off
  oil preheating: off
  heating activated: on
  heating temperature: 90.0 °C
  boiler pump max power: 100 %
  boiler pump min power: 100 %
  pump delay: 5
  burner min period: 10
  burner min power: 0 %
  burner max power: 100 %
  hysteresis on temperature: -6.0 °C
  hysteresis off temperature: 6.0 °C
  burner current power: 0 %
  burner starts: 137783
  total burner operating time: 463 days 2 hours 30 minutes
  burner stage 2 operating time: 0 days 0 hours 0 minutes
  total heat operating time: 403 days 10 hours 36 minutes
  total UBA operating time: 1912 days 15 hours 33 minutes
  last error code: 6L(512) 29.04.2022 00:55 (1 min)
  service code: 0H
  service code number: 0
  maintenance message: H00
  dhw set temperature: 60.0 °C
  dhw selected temperature: 60.0 °C
  dhw type: buffer
  dhw comfort: hot
  dhw flow temperature offset: 40.0 °C
  dhw circulation pump available: off
  dhw charging type: chargepump
  dhw hysteresis on temperature: -5.0 °C
  dhw hysteresis off temperature: -1.0 °C
  dhw disinfection temperature: 70.0 °C
  dhw circulation pump mode: 2x3min
  dhw circulation active: off
  dhw current intern temperature: 58.5 °C
  dhw current tap water flow: 0.0 l/min
  dhw storage intern temperature: 58.5 °C
  dhw activated: on
  dhw one time charging: off
  dhw disinfecting: off
  dhw charging: off
  dhw recharging: on
  dhw temperature ok: on
  dhw active: off
  dhw 3-way valve active: off
  dhw starts: 17829
  dhw active time: 59 days 17 hours 13 minutes

Thermostat: RC35 (DeviceID:0x10, ProductID:86, Version:01.08) (58)
  last error code: 5L(542) 04.02.2020 15:31 (1 min)
  date/time: 06.11.2022 19:22
  internal temperature offset: 0.0 °C
  minimal external temperature: -10.0 °C
  temperature sensor 1: 34.7 °C
  temperature sensor 2: 35.4 °C
  damping outdoor temperature: on
  damped outdoor temperature: 10.0 °C
  building type: heavy
  dhw mode: auto
  dhw circulation pump mode: auto
  dhw program: std prog
  dhw circulation program: std prog
  dhw disinfecting: off
  dhw disinfection day: tu
  dhw disinfection hour: 1
  dhw maximum temperature: 60.0 °C
  dhw one time key function: on
  dhw program switchtime: 00 mo 05:30 on
  dhw circulation program switchtime: 00 mo 05:30 on
  dhw holiday dates: 01.01.2000-01.01.2000
  dhw vacation dates: 01.01.2000-01.01.2000
  hc1 selected room temperature: 22.0 °C
  hc1 mode: auto
  hc1 mode type: day
  hc1 day temperature: 22.0 °C
  hc1 night temperature: 19.0 °C
  hc1 design temperature: 85.0 °C
  hc1 offset temperature: 2.0 °C
  hc1 holiday temperature: 10.5 °C
  hc1 target flow temperature: 5.0 °C
  hc1 summer temperature: 20.0 °C
  hc1 summer mode: winter
  hc1 holiday mode: off
  hc1 nofrost temperature: 5.0 °C
  hc1 nofrost mode: room
  hc1 room influence: 0.0 °C
  hc1 min flow temperature: 5.0 °C
  hc1 max flow temperature: 80.0 °C
  hc1 flow temperature offset for mixer: 5.0 °C
  hc1 heating type: radiator
  hc1 reduce mode: reduce
  hc1 control mode: outdoor
  hc1 control device: rc3x
  hc1 holiday dates: 01.01.2000-01.01.2000
  hc1 vacation dates: 19.06.2019-22.06.2019
  hc1 program: own 1
  hc1 pause time: 0 hours
  hc1 party time: 0 hours
  hc1 temporary set temperature automode: 0.0 °C
  hc1 no reduce below temperature: -31.0 °C
  hc1 off/reduce switch temperature: 5.0 °C
  hc1 vacations off/reduce switch temperature: 5.0 °C
  hc1 vacations reduce mode: room
  hc1 dhw priority: on
  hc1 own1 program switchtime: 00 mo 07:00 on
  hc1 own2 program switchtime: 00 mo 05:30 on

Controller: BC10/RFM20 (DeviceID:0x09, ProductID:68, Version:02.03) (0)
```

Support Information attached:
[emsesp_info.txt](https://github.com/emsesp/EMS-ESP32/files/9946698/emsesp_info.txt)

**Comment by proddy** [comment]
Firstly we need some system information (see https://emsesp.github.io/docs/#/Support?id=creating-an-new-request). Can you include that please? Also always upgrade to the latest version (now 3.5.0b9) first as perhaps some of the issues have been already resolved.

When you say missing, missing from what? Can you see these data points somewhere else we can use for reference?

The retTemp (return temperature) should be there if your boiler supports it, but I see it's missing so we'll need some telegrams to double-check. But first do the firmware upgrade. Same with SysPress (system pressure). 

PumpStatus is part of the Mixer device and not the Boiler but I didn't see that listed above?

What is valveStatus ?

**Comment by Marcwa19197** [comment]
I updated my first comment and updated to the latest version.

With missing i mean im not able to fetch the date via telegraf. So i also guess mqtt is not providing it.
Or is there a way to check the values directly on the ems-esp via telnet?

If i run "show mqtt" on the ems-esp i see the following:
```
MQTT is connected
MQTT publish errors: 0

MQTT topic subscriptions:
 ems-esp/system/#
 ems-esp/dallasssensor/#
 ems-esp/analogsensor/#
 ems-esp/boiler/#
 ems-esp/thermostat/#

MQTT queue is empty
```

So after the upgrade still no retTemp or SysPress available.
Yes, pumpStatus and valveStatus is provided by the mixer. But also im not able to see this data.
(Im trying to get the Grafana Dashboard working exactly as in the example).

Thanks.

**Comment by proddy** [comment]
I really need the support details. In the Web UI go to help and read the last bullet. I think your mixer is not being detected.

If you want to see what's being sent to your MQTT broker installed MQTTExplorer

**Comment by Marcwa19197** [comment]
I uploaded the info in the first post. thanks. Will check MQTTExplorer.

**Comment by proddy** [comment]
sorry, missed that. Is the LT160 a Mixer? The web and console both show all the data. If it's not there, it wouldn't be in the MQTT messages either.

**Comment by Marcwa19197** [comment]
No, LT160 is only the warmwater storage. I checked the it right now in the basement, but im not able to tell you what it is....
There is no label at all.

**Comment by MichaelDvP** [comment]
I have nearly the same system (GB125, MC10, RC35, MM10, SM10).
>    reTemp (Im able to see the curFlowTemp but not the Returning Temp)

The GB125 does not have sensor for returntemp, just connect a dallassensor. 
>    sysPress

Also no sensor, i use a analog pressure sensor. Make sure the analog signal does not exceed 1.5V.
>    pumpStatus

What pump? You have heattingpump and dhw charge pump, both are listed.
>    valveStatus (i have one Heating Cycle, so i subscribed to topic "mixer_data_hc1" and "mixer_data_ww1")

What valve? You do not have a mixer and dhw type is chargepump, so no 3-way-valve.
The picture shows the housing of heatingpump/header. Same housing is used for mixers, but with the lower window opend to see the mixing position.

**Comment by Marcwa19197** [comment]
Thank you for the information.
I also have a analog pressure sensor on the left side. What do you mean with 1.5V? And what do you meant exactly with an dallassensor? How can i install one on the system? I think at least the ReturnTemperature would be great to monitor.

Thank you for the explaination regarding the valveStatus! Its clear to me now! :-)

**Comment by MichaelDvP** [comment]
Do you have a electronic sensor or a gauge? 
I meant a cheap sensor like [this](https://www.ebay.de/itm/313886871476?hash=item49151d8fb4:g:9ikAAOSwHa1iFQXu&amdata=enc%3AAQAHAAAA4HsxU36%2Bz0F3AYKGionSUNWSsGHlzCpON4j64dJLT4cDhdqTJK8y%2Fo7wr2Nj4%2BCUl5C2yi%2By2gs8DCUOyahiFb3q8L9nFzoZUza0EnRBVNzS3SkajKx65cWXYdJZ4O8oK9Gz7IXiekuZADk%2B9wkdZnFHQD0zhrO%2Buh88KwQkP9qd%2Fu7wItfKzAvTdUPPHggJEaZQdWeWZuo5%2BUODuyONc5iHySeTyeaBdOWkaMCw9wiEUX1JNzvPLD4Rr0ukY7h3LKQ4PS6z2YkY1tjU62V58C5dRiVLscWmTXOlxMWT83rp%7Ctkp%3ABFBM6unHloph) avaible with different ranges 0.8/1.0/1.2 MPa. 5V can taken from the gateway, but 4.5V full range is to much for the esp.  You can selsct the sensor for your expected full range and use a voltage divider 3:1. Or us a sensor with higher range connected directly. With normal pressure in heating of 1,5-2 bar a 1MPa sensor is ok, the output is 1.5 V at 2,5 bar pressure.
Dallassensor means [DS18B20](https://www.ebay.de/itm/255283194278?hash=item3b701069a6:g:dcEAAOSwXatgskBd&amdata=enc%3AAQAHAAAA4F7RZ3FQ1bhHIdTmghvjDmXs0KCW3LJEhq9yu9vmSljCEgIjVDio%2F1udOP7GNNxeLGzX%2Bh8lGOj%2B2hrsvcqfpur%2FAOqp%2FnQJ1qNfz6QIF79c8sOMQ27M5Ts94kI0gHGogNhWFb4%2Bl20dKdiIOG6bXF4%2Fq4cWnjZ8U4CyMlhxSZXnP7NTZircTMbe1rFYGvmv1K5x%2B91rsqp94%2BUdcLkKUzzPBIGD7NL7o7NnC4sa9Iy%2BvMUuoVnjtsWAjacEwO7%2Fowe2Agn%2FNhmd9%2FySkkXPHRcHazqD1%2BYrorvPSudFFuhm%7Ctkp%3ABk9SR8KRtJeKYQ), that can be connected directly to the gateway.

**Reply by Marcwa19197** [reply]
> Do you have a electronic sensor or a gauge? I meant a cheap sensor like [this](https://www.ebay.de/itm/313886871476?hash=item49151d8fb4:g:9ikAAOSwHa1iFQXu&amdata=enc%3AAQAHAAAA4HsxU36%2Bz0F3AYKGionSUNWSsGHlzCpON4j64dJLT4cDhdqTJK8y%2Fo7wr2Nj4%2BCUl5C2yi%2By2gs8DCUOyahiFb3q8L9nFzoZUza0EnRBVNzS3SkajKx65cWXYdJZ4O8oK9Gz7IXiekuZADk%2B9wkdZnFHQD0zhrO%2Buh88KwQkP9qd%2Fu7wItfKzAvTdUPPHggJEaZQdWeWZuo5%2BUODuyONc5iHySeTyeaBdOWkaMCw9wiEUX1JNzvPLD4Rr0ukY7h3LKQ4PS6z2YkY1tjU62V58C5dRiVLscWmTXOlxMWT83rp%7Ctkp%3ABFBM6unHloph) avaible with different ranges 0.8/1.0/1.2 MPa. 5V can taken from the gateway, but 4.5V full range is to much for the esp. You can selsct the sensor for your expected full range and use a voltage divider 3:1. Or us a sensor with higher range connected directly. With normal pressure in heating of 1,5-2 bar a 1MPa sensor is ok, the output is 1.5 V at 2,5 bar pressure. Dallassensor means [DS18B20](https://www.ebay.de/itm/255283194278?hash=item3b701069a6:g:dcEAAOSwXatgskBd&amdata=enc%3AAQAHAAAA4F7RZ3FQ1bhHIdTmghvjDmXs0KCW3LJEhq9yu9vmSljCEgIjVDio%2F1udOP7GNNxeLGzX%2Bh8lGOj%2B2hrsvcqfpur%2FAOqp%2FnQJ1qNfz6QIF79c8sOMQ27M5Ts94kI0gHGogNhWFb4%2Bl20dKdiIOG6bXF4%2Fq4cWnjZ8U4CyMlhxSZXnP7NTZircTMbe1rFYGvmv1K5x%2B91rsqp94%2BUdcLkKUzzPBIGD7NL7o7NnC4sa9Iy%2BvMUuoVnjtsWAjacEwO7%2Fowe2Agn%2FNhmd9%2FySkkXPHRcHazqD1%2BYrorvPSudFFuhm%7Ctkp%3ABk9SR8KRtJeKYQ), that can be connected directly to the gateway.

Hi Michael, 
i recently got some spare time so i purchased a sensor you mentioned with 80 PSI on ebay (https://www.ebay.de/itm/405430490874?_trkparms=amclksrc%3DITM%26aid%3D111001%26algo%3DREC.SEED%26ao%3D1%26asc%3D279998%26meid%3D610580786072423e98595cc22192724b%26pid%3D102020%26rk%3D1%26rkt%3D15%26sd%3D255283194278%26itm%3D405430490874%26pmt%3D1%26noa%3D1%26pg%3D2332490%26brand%3DMarkenlos&_trksid=p2332490.c102020.m5276). But if i connect it to the gateway, together with my 5 dallas temperature-sensors no sensor is detected anymore. did i purchased the wrong one? I also read on the documentation, that i can only use the same model / sensors in parallel, is that correct?

How can i use a pressure-sensor in parallel with my dallas-temperature sensors? 

Thank you very much for a reply and also for the great product and software, which is running since years without problems!

**Reply by MichaelDvP** [reply]
This is a 0,5-4,5V output sensor. You need to supply the sensor with 5V and connect the signal to a free gpio.
80psi is ~ 5,5bar full range (@4.5V). The analog input of ems-esp is limited to ~1,5V, so you can measure only up to 1,4 bar. Check if this is ok with your system pressure. 
If you like to compile your own version you can set a different value to the attentuation here: https://github.com/emsesp/EMS-ESP32/blob/4151a82b3b0500e91dab23556bdfb3ca165b4d57/src/core/analogsensor.cpp#L33, e.g. ADC_11db for 2,8V full scale (3,4bar measuring range),
When adding the anlogsensor use ADC with offset 500 mV and factor 0,00138 for reading in bar.

**Reply by Marcwa19197** [reply]
Thanks for the reply! I need a pressure sensor till 2.0 / 2.5bar maximum. My system pressure is 2.0.
So i guess there is no chance using a analog sensor connected "normally" not to the internal gpio? Also i need a self compiled version?

**Reply by MichaelDvP** [reply]
> So i guess there is no chance using a analog sensor connected "normally" not to the internal gpio? 

?

Analog sensors always connect to internal gpio and has to be configured in analog sensor menu. You should only connect one if you're sure what you're doing. I recommended a 1MPa sensor, which gives a usable range of ~2,5bar. You've get a 80psi => 0,5MPa sensor, so you have half the range. You still can add an external voltage divider and adapt the range (or set the internal attentuator to different value and compile your own software). But first make sure you know what au're doing.

**Comment by pudermueller** [comment]
> No, LT160 is only the warmwater storage. I checked the it right now in the basement, but im not able to tell you what it is.... There is no label at all. ![IMG_0394](https://user-images.githubusercontent.com/5463753/200192699-13f0237e-3854-4345-b827-8fd52fae9608.jpeg)

This is the mixer of your heating circuit. This includes also your heat circuit pump you are missing in MQTT. 

Concerning MQTT you should sniff with MQTT Explorer. It should show you a "ems-esp/mixer_data_hc2" <-- 2 and not 1. My GB135 has the heating circuit 1 only for a configuration without mixer and separate pump. If you have a mixer this can in RC35 only be linked to HC2. Same is valid for the valvestatus (also in mixer_data_hc2). 

Regards
Pudermueller

**Comment by Marcwa19197** [comment]
Hi @pudermueller ,
thanks for the reply.

I sniffed with MQTT Explorer and im not able to see something mixer related. Im a bit confused now. Is it because of missing hardware? And if so, what im missing? I guess it woul be better to have the returntemp available, so the GB125 "knows" how much it must reheat the heating cycle, or? 

<img width="515" alt="image" src="https://user-images.githubusercontent.com/5463753/201512929-052ce460-ddc9-4ec7-a972-83e5c8d922bb.png">

**Comment by pudermueller** [comment]
Hi @Marcwa19197 , 
@MichaelDvP  already identified correctly that your unit is not a mixer unit. You can remove the cover by opening the black plastic screws and lift it. For comparison you might check [Buderus flyer](https://productsde.buderus.com/buderus/productsde.buderus.com/broschueren/broschuere-buderus-heizkreisset-25-6-msl-de-052020.pdf). 
Coming back to your original request: you are able to see the pump status (heating circuit and warm water production) 
and for return temperature you need an additional dallassensor. 
Best regards Pudermueller

**Comment by Marcwa19197** [comment]
Hello again :-) Thank you very much!
So during the Winter i installed two Dallassensors to get return and also in addition to get forward temperatures directly from the pipes.

Im checking if i somehow can install a pressure sensor like @MichaelDvP mentioned.
@MichaelDvP How did you install it on your system? Do you have the same setup like in my picture on https://github.com/emsesp/EMS-ESP32/discussions/986#discussioncomment-4954734?
